src/main/common/utils.h

```diff
@@ -55,7 +55,7 @@
 #define PP_CALL(macro, ...) macro(__VA_ARGS__)
 
 #if !defined(UNUSED)
-#define UNUSED(x) (void)(x) // Variables and parameters that are not used
+#define UNUSED(x) ((void)(x)) // Variables and parameters that are not used
```

Comment:
Not sure that this is the right solution, as this merely hides the problem. This approach will break if another library that defines unused as `(void)(x)` is added.
What we have done in the past is to isolate the definition in the library with a `#if !defined(UNUSED)`.

Reply:
I originally added the `#if !defined(` to the library, but my concern was trying to avoid changes to the library if possible. Looking at it deeper there are a number of other places it's defined in the libraries with and without the extra parenthesis.
```
lib/main/STM32_USB_Device_Library/Class/msc/src/usbd_msc_bot.c:#define UNUSED(x) (void)(x)
lib/main/STM32_USB_Device_Library/Class/msc/src/usbd_msc_core.c:#define UNUSED(x) (void)(x)
lib/main/STM32_USB_Device_Library/Class/msc/src/usbd_msc_scsi.c:#define UNUSED(x) (void)(x)
lib/main/STM32F1/Drivers/STM32F1xx_HAL_Driver/Inc/stm32f1xx_hal_def.h:#define UNUSED(x) ((void)(x))
lib/main/STM32F3/Drivers/STM32F3xx_HAL_Driver/Inc/stm32f3xx_hal_def.h:#define UNUSED(x) ((void)(x))
lib/main/STM32F4/Drivers/STM32F4xx_HAL_Driver/Inc/stm32f4xx_hal_def.h:#define UNUSED(x) ((void)(x))
lib/main/STM32F7/Drivers/STM32F7xx_HAL_Driver/Inc/stm32f7xx_hal_def.h:#define UNUSED(x) ((void)(x))
```
So now I'm not sure there's a good solution without changing the libraries to block them all with `#if !defined(`.

Reply:
Leave it to STM to be inconsistent in the stuff they give us. :confounded: